src/host/posix/dnssd.hpp

```diff
@@ -31,11 +31,13 @@
  *   This file includes definitions for implementing OpenThread DNS-SD platform APIs.
  */
 
-#ifndef OTBR_AGENT_POSIX_DNSSD_HPP_
-#define OTBR_AGENT_POSIX_DNSSD_HPP_
+#ifndef OTBR_POSIX_DNSSD_PLAT_HPP_
```

Comment:
I'm not sure if we should add the `PLAT` into this macro because it's not in the name of the header file.